src/Native/Unix/System.Native/pal_dl.cpp

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <dlfcn.h>
+
+extern "C" void* SystemNative_DlOpen(const char *file, int mode)
```

Comment:
Generally we try to avoid just assuming that these values are the same on all platforms, e.g. that RTLD_GLOBAL is 4 everywhere.  Two options:
1. Switch on mode, accepting the values we use in .NET, and use that to determine the actual value to pass to dlopen.
2. Use static_asserts to validate the values we're using for mode match the platform at compile time.

Reply:
Thanks, I limited the values to `RTLD_LAZY ` and `RTLD_NOW`, added `PAL_` equivalents and added `static_asserts` to validate the values at compile time.